tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -195,22 +200,63 @@ sub enable_vf {
         push @drivers, $driver if (!grep /^$driver$/, @drivers);
     }
 
+    #change to use sol console to reload network devcie drivers as ssh connection will break
+    select_console('sol', await_console => 0);
+    send_key 'ret' unless (check_screen([qw(text-login text-logged-in-root)], 20));    #sometimes or on some machines, 'ret' is required
+    if (check_screen("text-login", 10)) {                                              #login is required for the first access to sol console
+        type_string "root\n";
+        assert_screen "password-prompt";
+        type_password;
+        send_key 'ret';
+    }
+    assert_screen "text-logged-in-root";
+    send_key 'ret';
+
     #set max_vfs and reload driver
     #should not enable vf repeatedly, so skip enabling in local test for debugging
     foreach my $driver (@drivers) {
-        assert_script_run("[ `lsmod | grep $driver | wc -l` -gt 0 ] && rmmod $driver", 60);
-        assert_script_run("modprobe --first-time $driver max_vfs=8",                   60);
+        type_string("[ `lsmod | grep $driver | wc -l` -gt 0 ] && rmmod $driver\n");
+        sleep 30;
+        type_string("modprobe --first-time $driver max_vfs=8\n");
+        save_screenshot;
+        sleep 30;
```

Comment:
You can use type_string(timeout=>)

Reply:
Ipmi sol console part is removed as it has not been needed anymore. But your advice is still helpful for my future test codes.

Reply:
I never used type_string with timeout either. Anyway I think you can use assert_script_run(timeout, proceed_on_failure)

Reply:
assert_script_run() does not always work in ipmi sol console, including script_output() and script_run(). type_string() is more stable than them.

Reply:
Choose the one suits you best.